client/extensions/woocommerce/app/products/product-variation-types-form.js

```diff
@@ -72,7 +86,20 @@ class ProductVariationTypesForm extends Component {
 			find( product.attributes, function( a ) {
 				return a.uid === attributeId;
 			} );
-		editProductAttribute( siteId, product, attribute, { name } );
+
+		// Ensure we don't have an existing variation type with the same name.
+		const existingAttribute =
+			product.attributes &&
+			find( product.attributes, function( a ) {
+				return a.uid !== attributeId && a.name.toLowerCase() === name.toLowerCase();
```

Comment:
Good call on the casing before comparison. Please also consider trimming before comparison too. (We probably should be trimming these before persisting anyways.